### PR TITLE
composer.json compatible with Composer 2.2

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,6 +12,11 @@
     "config": {
         "platform": {
             "php": "8.0.99"
+        },
+        "allow-plugins": {
```
Comment (Ocramius) [inline composer.json:16]: Oh, composer also broke compat here? :| 

I'm starting to see frikken BC breaks everywhere...

File: composer.json [hunk 2]
```diff
@@ -12,6 +12,11 @@
...
+            "dealerdirect/phpcodesniffer-composer-installer": true,
+            "infection/extension-installer": true,
```
Comment (Ocramius) [inline composer.json]: Can be removed: we don't use this

File: composer.json [hunk 3]
```diff
@@ -12,6 +12,11 @@
...
+            "composer/package-versions-deprecated": true
```
Comment (Ocramius) [inline composer.json]: We should depend on `ocramius/package-versions:^2`, and not on `composer/package-versions-deprecated`, which still uses the problematic approach of `ocramius/package-versions:^1` (long gone)

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json [hunk 4]
```diff
@@ -12,6 +12,10 @@
     "config": {
         "platform": {
             "php": "8.0.99"
+        },
+        "allow-plugins": {
+            "dealerdirect/phpcodesniffer-composer-installer": true,
+            "infection/extension-installer": false
```
Comment (Ocramius) [inline composer.json:18]: Should still not be necessary :thinking:

Comment (kukulich) [reply composer.json:18]:
```
$ composer install
dealerdirect/phpcodesniffer-composer-installer contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
Do you trust "dealerdirect/phpcodesniffer-composer-installer" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?] y
infection/extension-installer contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
Do you trust "infection/extension-installer" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?] n
```

🤷‍♂️

Comment (kukulich) [reply composer.json:18]: I can only say `y` or `n` to get rid of the question once for all.

Comment (Ocramius) [reply composer.json:18]: Can we add a `--no-interaction` or such, perhaps?

Comment (kukulich) [reply composer.json:18]:
```
For additional security you should declare the allow-plugins config with a list of packages names that are allowed to run code. See https://getcomposer.org/allow-plugins
You have until July 2022 to add the setting. Composer will then switch the default behavior to disallow all plugins.
```

Btw `--no-interaction` is in all workflows. I just want to get rid of it on my localhost :)

Comment (Ocramius) [reply composer.json:18]: Eh, for localhost you'll have to suffer this pain :D 

Alternatively, we start isolating all this stuff in `Makefile` definitions :thinking:

Comment (kukulich) [reply composer.json:18]: I'm happy Windows user so I'm not sure what will cause me more suffering 😄

Comment (Ocramius) [reply composer.json:18]: I'll be honest: that's you looking for trouble :P

File: composer.json (hunk 1 above)
Comment (WyriHaximus) [reply composer.json:16]: Wait hold up, we have to allow plugins in every package now?

**Pull request closed without merging**